### ISSUE TITLE
can we have a docker-compose.yml with full running environment

### ISSUE THREAD

**Issue by ozbillwang** [Unicon/shibboleth-idp-dockerized#24]
I follow this [document](https://blogs.aws.amazon.com/security/post/TxMPIN2JKCL9NT/New-Whitepaper-Single-Sign-On-Integrating-AWS-OpenLDAP-and-Shibboleth), then google leads me to this open source repo.

Seems it can save my time to set it.

I do follow up the README, and customize `shibboleth-idp` setting. But I still can't make it work with my ldap server. 

The error I got is below, seems it try to access port `8080` which is not documented. 

```
2016-07-12 13:57:16.978:INFO:oejs.ServerConnector:main: Started ServerConnector@38ee7a9d{SSL,[ssl, http/1.1]}{0.0.0.0:8443}
2016-07-12 13:57:16.979:INFO:oejs.ServerConnector:main: Started ServerConnector@629984eb{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}
Exception in thread "main" java.lang.IllegalStateException: no valid keystore
    at org.eclipse.jetty.util.security.CertificateUtils.getKeyStore(CertificateUtils.java:48)
```

So could we have a running `docker-compose.yml`, which include ldap + shibboleth, that guarantee to have running docker containers directly?

**Comment by jtgasper3** [comment]
Port 8080 is just what is opened by default and it's impact is minimal. The error is related to a keystore, probably the one associated with the backchannel?... That's my guess.

Here's a full working IdP (based off of this image), LDAP, and several SPs. https://github.com/UniconLabs/dockerized-idp-testbed. 

Hopefully that will give you the missing clue.

**Comment by ozbillwang** [comment]
Finally get the problem fixed with help from issue #23 

https://github.com/Unicon/shibboleth-idp-dockerized#browser-based-tls-certificate-and-key
